### PR TITLE
Add webmock for request stubbing during tests

### PR DESCRIPTION
**Pull request by dunkOnIT**
This will be used in #8927, but introducing it as a separate PR given that its out of scope for that PR. 
- [x] Explciitly enable net connect
- [x] Require webmock in spec_helper

### CODE REVIEW

**Review by gregorbg — CHANGES_REQUESTED** [review]
Please add an evnrionment config (in the `.env` files) that allows you to enable/disable webmock by setting the corresponding env var to true/false (and have the spec config files pick up the env variable accordingly)

I also think that the default should be to disable webmock. and let developers choose locally whether they would like to enable it for local testing. The reasonable default should be to make live calls wherever and whenever possible

**Comment by dunkOnIT** [comment]
Added env variables for the following:
* enabling/disabling HTTP connections
* enabling/disabling webmock stubs (not used in this PR, but will be incorporated into the Paypal PR)

**Comment by gregorbg** [comment]
You famously missed an `end` in your latest ommit. The tests cannot even start to run because of syntax errors.

**Review by gregorbg — APPROVED** [review]

File: env_config.rb [hunk 1]
```diff
@@ -44,10 +44,15 @@
 
     # Local-specific stuff
     optional :ENABLE_BULLET, :bool, false
-    optional :SKIP_PRETEST_SETUP, :bool, false
     optional :MAILCATCHER_SMTP_HOST, :string, ''
   end
 
+  if Rails.env.test?
+    optional :DISABLE_WEBMOCK, :bool, false
+    optional :ENABLE_NET_CONNECT_IN_TESTS, :bool, true
```
Comment (gregorbg) [inline env_config.rb:52]: Here you declared it as `ENABLE` but below in the actual config you're using `DISABLE`

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -87,6 +87,12 @@
   config.include ApplicationHelper
 
   config.include ActiveJob::TestHelper
+
+  if EnvConfig.DISABLE_WEBMOCK?
+    WebMock.disable!
+  else
+    WebMock.allow_net_connect! unless EnvConfig.DISABLE_NET_CONNECT_IN_TESTS?
```
Comment (gregorbg) [inline spec/rails_helper.rb:94]: ENABLE vs DISABLE, see above.

**Review by gregorbg — CHANGES_REQUESTED** [review]

**Review by gregorbg — APPROVED** [review]

**Pull request merged**